src/main/java/org/jabref/logic/importer/fetcher/WorldcatFetcher.java

```diff
@@ -0,0 +1,95 @@
+package org.jabref.logic.importer.fetcher;
+
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import org.jabref.logic.importer.EntryBasedFetcher;
+import org.jabref.logic.importer.FetcherException;
+import org.jabref.logic.importer.ParserResult;
+import org.jabref.logic.importer.fileformat.WorldcatImporter;
+import org.jabref.logic.net.URLDownload;
+import org.jabref.model.database.BibDatabase;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.field.StandardField;
+
+/**
+ * EntryBasedFetcher that searches the Worldcat database
+ * 
+ * @see https://www.oclc.org/developer/develop/web-services/worldcat-search-api/bibliographic-resource.en.html
+ */
+public class WorldcatFetcher implements EntryBasedFetcher {
+
+    public static String API_KEY;
+    private static final String NAME = "Worldcat Fetcher";
+
+    private static final String WORLDCAT_OPEN_SEARCH_URL = "http://www.worldcat.org/webservices/catalog/search/opensearch?wskey=" + API_KEY;
+
+    public WorldcatFetcher (String worldcatKey) {
+        API_KEY = worldcatKey;
```

Comment:
Codacy complains here: "Possible unsafe assignment to a non-final static field in a constructor.". I think, it's right and the static variable in line 29 should be replaced by an instance variable

Check https://app.codacy.com/gh/JabRef/jabref/pullRequest?prid=5083307#newIssuesView for details.